draco/tests/test_soft.py

```diff
@@ -0,0 +1,38 @@
+from draco.asp_utils import Block
+from draco.programs import soft
+from draco.tests.test_hard import list_violations, no_violations
+
+
+def test_aggregate():
+    b = soft.blocks["aggregate"]
+    assert isinstance(b, Block)
+
+    assert no_violations(
+        b.program
+        + """
+    attribute((mark,type),m1,text).
+    entity(encoding,m1,e1).
+    attribute((encoding,channel),e1,text).
+    """
+    )
+
+    assert (
+        list_violations(
+            b.program
+            + """
+    attribute((encoding,aggregate),e1,mean).
+    """
+        )
+        == ["aggregate"]
+    )
+
+    assert (
+        list_violations(
+            b.program
+            + """
+    attribute((encoding,aggregate),e1,mean).
+    attribute((encoding,aggregate),e2,mean).
+    """
+        )
+        == ["aggregate"] * 2
```

Comment:
Great to count it twice. I wonder whether we should figure out a way to return the whole tuple, not just the name of the violation? This way we would know where the violation is happening. Could return for example `("aggregate", "e1")`.